noxfile.py

```diff
@@ -58,21 +67,41 @@ def _contrib(session):
         return
 
     if session.posargs:
-        session.run("pytest", *session.posargs)
+        session.run(*runner_commands, *session.posargs)
     else:
         # defaults
+        default_runner_commands = runner_commands.copy()
+        if "--append" in default_runner_commands:
+            default_runner_commands.pop(default_runner_commands.index("--append"))
```

Comment:
Perform a copy here so that when `--append` is `pop`ed from the list it doesn't get `pop`ed from the `posargs` list that will be used in `_contrib` and so then won't clobber the previous `.coverage` file from the rest of the tests.